libs/build-nss-desktop.sh

```diff
@@ -62,8 +62,8 @@ if [[ "${CROSS_COMPILE_TARGET}" =~ "darwin" ]]; then
     # run-task has already downloaded + extracted the dependency
     NSS_DIST_DIR="${MOZ_FETCHES_DIR}/dist"
   else
-    curl -sfSL --retry 5 --retry-delay 10 -O "https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/app-services.cache.level-3.content.v1.nss-artifact.latest/artifacts/public/dist.tar.bz2"
```

Comment:
Looks like we're switching here from using our own taskcluster index to a specific run's artifacts, is this intentional? I vaguely assumed that we now pull from the CI in the `fetch` taskcluster job, then upload it here, then we later just use it from our index?

I'm out of my depth here so I could be very off 😅

Reply:
Yeah this will be switched to our L3 cache, but I think it's more clear to just use NSS TC since it's the same as CI right now then update to use the L3 cache once this lands. I feel our docs are not as clear on the best approach here and will make a task to update that.